app/controllers/mixins/breadcrumb_mixin.rb

```diff
@@ -0,0 +1,151 @@
+module Mixins
+  module BreadcrumbMixin
+    def self.included(c)
+      c.helper_method(:data_for_breadcrumbs)
+    end
+
+    def data_for_breadcrumbs
+      if defined?(breadcrumbs_options)
+        breadcrumbs = breadcrumbs_options[:breadcrumbs]
+        record_info = breadcrumbs_options[:record_info] unless breadcrumbs_options[:record_info].nil? || breadcrumbs_options[:record_info]["id"].nil?
+        show_list_title = breadcrumbs_options[:show_list_title]
+        include_record = breadcrumbs_options[:include_record]
+      end
+
+      breadcrumbs = breadcrumbs.nil? ? [] : breadcrumbs
+
+      # Different methods for controller with explorers and for non-explorers controllers
+
+      if @sb[:explorer].blank? && @right_cell_text.blank?
+        breadcrumbs = build_breadcrumbs_no_explorer(breadcrumbs, record_info, show_list_title)
+        breadcrumbs = build_title_breadcrumb(breadcrumbs)
```

Comment:
These two lines just don't make sense together, the second one invalidates the first and you're rewriting a variable that's on the right side, that makes the code less readable.